src/plugins/chart_expressions/expression_xy/public/helpers/data_layers.tsx

```diff
@@ -199,13 +198,42 @@ export const getFormattedTablesByLayers = (
     {}
   );
 
-const getSeriesName: GetSeriesNameFn = (
+function getSplitValues(
+  splitAccessorsMap: XYChartSeriesIdentifier['splitAccessors'],
+  splitAccessors: Array<string | ExpressionValueVisDimension>,
+  alreadyFormattedColumns: Record<string, boolean>,
+  columns: Datatable['columns'],
+  formatFactory: FormatFactory
+) {
+  if (splitAccessorsMap.size < 0) {
+    return [];
+  }
+
+  return [...splitAccessorsMap].reduce<Array<string | number>>((acc, [splitAccessor, value]) => {
+    const split = splitAccessors.find(
+      (accessor) => getAccessorByDimension(accessor, columns) === splitAccessor
+    );
+    if (split) {
+      const splitColumnId = getAccessorByDimension(split, columns);
+      const splitFormatByAccessor = getFormat(columns, split);
+      const splitFormatter = formatFactory(splitFormatByAccessor);
```

Comment:
This function will be called once per data point - as creating a field formatter is expensive we need to move this to the top level renderer so it's only happening once per split accessor, then pass down the array of formatters and access it here.

Reply:
done